torch/distributions/uniform.py

```diff
@@ -33,6 +36,10 @@ def __init__(self, low, high):
             batch_shape = self.low.size()
         super(Uniform, self).__init__(batch_shape)
 
+    @property
+    def support(self):
+        return constraints.interval(self.low, self.high)
```

Comment:
This is a bit worrying for me. `support` and `params` are usually class attributes, except for some edge cases. This means that `Distribution.params[name]` is usually well defined, but not always, and there's no clear rule to tell when it is. How about making them always be present on a class, but be `dependent` at that time, but guarantee that they will be resolved to concrete values if you instantiate a distribution?

Reply:
Done.  (I didn't think this was possible until you suggested it :smile:)